tensorflow/python/ops/math_ops.py

```diff
@@ -4857,6 +4857,68 @@ def sparse_segment_sum(
         name=name,
     )
 
+@tf_export("sparse.sampled_addmm")
+def sampled_addmm(indices,
+    values,
+    dense_shape,
+    mat1,
+    mat2,
+    beta=1.0,
+    alpha=1.0
+):
+  sparse_rank = len(dense_shape)
```

Comment:
This won't always work if `dense_shape` is a dynamic shape obtained from a tensor, e.g. via `tf.shape`.

Reply:
I guess we could probably assume the dense shape based on the shape of the product of the matrix multiplication. So we could probably get rid of this `dense_shape` input altogether

Reply:
You may want to verify they are the same.  The input here is essentially a `SparseTensor` broken down into components.  I would keep it as is.